src/pseudopeople/interface.py

```diff
@@ -77,19 +81,24 @@ def _generate_dataset(
 
     # Known pandas bug: pd.concat does not preserve category dtypes so we coerce
     # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
-    noised_dataset = _coerce_dtypes(noised_dataset, dataset)
+    noised_dataset = _coerce_dtypes(noised_dataset, dataset, cleanse_int_cols=True)
 
     logger.debug("*** Finished ***")
 
     return noised_dataset
 
 
-def _coerce_dtypes(data: pd.DataFrame, dataset: Dataset):
+def _coerce_dtypes(
+    data: pd.DataFrame, dataset: Dataset, cleanse_int_cols: bool = False
+) -> pd.DataFrame:
     # Coerce dtypes prior to noising to catch issues early as well as
     # get most columns away from dtype 'category' and into 'object' (strings)
     for col in dataset.columns:
+        if cleanse_int_cols and col.name in INT_COLUMNS:
```

Comment:
I have to special case INT_COLUMNS here because on some of our integration tests we compare values to unnoised vs noised data which would then compare strings to non-string values. None of our tests compare these columns. It seems like there would be a better way to handle this but this is what I have working currently for the sake of time.